doc/man3/EVP_PKEY_sign.pod

```diff
@@ -96,6 +189,152 @@ Sign data using RSA with PKCS#1 padding and SHA256 digest:
 
  /* Signature is siglen bytes written to buffer sig */
 
+=head2 RSA-SHA256 with a pre-computed digest
+
+Sign a digest with RSA-SHA256 using one-shot functions.  To be noted is that
+RSA-SHA256 is assumed to be an implementation of C<sha256WithRSAEncryption>,
+for which the padding is pre-determined to be B<RSA_PKCS1_PADDING>, and the
+input digest is assumed to have been computed using SHA256.
+
+ #include <openssl/evp.h>
+ #include <openssl/rsa.h>
+
+ EVP_PKEY_CTX *ctx;
+ /* md is a SHA-256 digest in this example. */
+ unsigned char *md, *sig;
+ size_t mdlen = 32, siglen;
+ EVP_PKEY *signing_key;
+
+ /*
+  * NB: assumes signing_key and md are set up before the next
+  * step. signing_key must be an RSA private key and md must
+  * point to the SHA-256 digest to be signed.
+  */
+ ctx = EVP_PKEY_CTX_new(signing_key, NULL /* no engine */);
+ alg = EVP_SIGNATURE_fetch(NULL, "RSA-SHA256", NULL);
+
+ if (!ctx)
+     /* Error occurred */
+ if (EVP_PKEY_sign_init_ex2(ctx, alg, NULL) <= 0)
+     /* Error */
+
+ /* Determine buffer length */
+ if (EVP_PKEY_sign(ctx, NULL, &siglen, md, mdlen) <= 0)
+     /* Error */
+
+ sig = OPENSSL_malloc(siglen);
+
+ if (!sig)
```

Comment:
nit: should our coding style be followed here?  i.e. sig != NULL?

Reply:
Fixed them all